src/ray/core_worker/reference_count.cc

```diff
@@ -832,9 +832,12 @@ void ReferenceCounter::AddNestedObjectIdsInternal(
       // contained in the outer object ID so we do not GC the inner objects
       // until the outer object goes out of scope.
       for (const auto &inner_id : inner_ids) {
-        it->second.contains.insert(inner_id);
         auto inner_it = object_id_refs_.find(inner_id);
-        RAY_CHECK(inner_it != object_id_refs_.end());
```

Comment:
@stephanie-wang  sometimes this check fails. could this be a race condition we should handle? (like add it as borrower)

Reply:
Hmm yeah I think this check is probably too strong. Should be fine to remove it and insert an empty entry if one does not exist yet.